### ISSUE TITLE
Redundant nested array rendered when using @SchemaProperty(array = ...) with object schemas

### ISSUE THREAD

**Issue by crizzis** [swagger-api/swagger-core#4778]
`@SchemaProperty(name = "create", array = @ArraySchema(schema = @Schema(implementation = String.class)))`

renders as: 

```
"create": {
            "type": "array",
            "items": {
              "type": "string"
            }
 }
```

`@SchemaProperty(name = "create", schema = @Schema(implementation = MyCustomDto.class))`

renders as: 

```
"create": {
            "$ref": "#/components/schemas/MyCustomDto"
          }
```

All of the above to be expected. 

**However**, `@SchemaProperty(name = "create", array = @ArraySchema(schema = @Schema(implementation = MyCustomDto.class)))`

renders as: 

```
"create": {
            "type": "array",
            "items": {
              "type": "array",
              "items": {
                "$ref": "#/components/schemas/MyCustomDto"
              }
            }
          }
```

In other words, it would seem there is no way to obtain a flat array property with POJO elements, using `@SchemaProperty`. It's either a plain object schema or a nested array, no in-between.

**Comment by crizzis** [comment]
Update: I found a workaround: apparently `@SchemaProperty(name = "create", array = @ArraySchema(schema = @Schema(ref = "#/components/schemas/MyCustomDto")))` works just fine. 

The problem only appears when using `@Schema.implementation` instead.
